### PR TITLE
chore(docs-spec): Add `write_mode`, `migrate_mode` clarifications

### PR DESCRIPTION
**Pull request by erezrokah**
#### Summary

Related to https://github.com/cloudquery/cloudquery/issues/12526#issuecomment-1651384771

### CODE REVIEW

File: website/pages/docs/reference/destination-spec.md [hunk 1]
```diff
@@ -66,13 +66,16 @@ Specifies the update method to use when inserting rows. The exact semantics depe
 - `overwrite`: Same as `overwrite-delete-stale`, but doesn't delete stale rows from previous `sync`s.
 - `append`: Rows are never overwritten or deleted, only appended.
 
+> Switching from `overwrite-delete-stale` or `overwrite` to `append`, or from `append` to `overwrite-delete-stale` or `overwrite` is not supported without dropping all the tables.
```
Comment (disq) [inline website/pages/docs/reference/destination-spec.md:69]: 'all' only if the user has `tables: *`, no? One can theoretically sync different tables with separate/incompatible modes as long as they are not synced in the same run

**Review by hermanschaaf — APPROVED** [review]

File: website/pages/docs/reference/destination-spec.md (hunk 1 above)
Comment (hermanschaaf) [reply website/pages/docs/reference/destination-spec.md:69]: Oh sorry got bit by automerge, but I think "all tables matched by your table config" is the most obvious interpretation of the word `all` here

Comment (hermanschaaf) [reply website/pages/docs/reference/destination-spec.md:69]: Maybe we can change it to what I said just to be clear we won't drop other tables though

Comment (erezrokah) [reply website/pages/docs/reference/destination-spec.md:69]: Good point, follow up in https://github.com/cloudquery/cloudquery/pull/12532

**Pull request merged**